pkg/storage-init/storage-init.sh

```diff
@@ -34,6 +34,57 @@ fi
 
 P3_FS_TYPE="ext3"
 FSCK_FAILED=0
+
+# First lets see if we're missing P3 altogether and try to create it.
+# This is safe, since the worst case scenario we may hose the system,
+# but the system without P3 is a warm brick anyways
+P3=$(/hostfs/sbin/findfs PARTLABEL=P3)
+IMGA=$(/hostfs/sbin/findfs PARTLABEL=IMGA)
+if [ -z "$P3" ] && [ -n "$IMGA" ]; then
+   DEV=$(echo /sys/block/*/"${IMGA#/dev/}")
+   DEV="/dev/$(echo "$DEV" | cut -f4 -d/)"
+
+   # if sgdisk complains we need to repair the GPT
+   if sgdisk -v "$DEV" | grep -q 'Identified.*problems'; then
```

Comment:
Dies all of this belong inside the "if persist partition doesn't exist" logic? Isn't this a completely independent piece? Ideally, it should be its own PR, but either way, just make a fn that runs before this called `validateMBR()` or similar?

Reply:
It does belong there in a sense, that I'm uncomfortable (still!) to touch GPT unless there's something as fundamentally broken as missing P3 or IMGB. So I'll keep it here for now if you don't mind -- but we can consider refactoring it later (especially if we do a more proper implementation of disk resizing along the lines of https://github.com/linuxkit/linuxkit/tree/master/pkg/format)